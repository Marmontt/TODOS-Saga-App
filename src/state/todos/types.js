export const ADD_TODO = 'TODO/ADD';
export const SET_FULFILLED_STATE = 'TODO/FULFILLED/STATE';

export const UPDATE_TODO = "TODO/UPDATE";
export const GET_TODOS_REQUEST = "TODOS/GET/REQUEST";
export const GET_TODOS_SUCCESS ="TODOS/GET/SUCCESS";

export const ADD_TODO_SUCCESS = 'TODO/ADD/SUCCESS';
export const SET_FULFILLED_SUCCESS = 'TODO/FULFILLED/SUCCESS';

