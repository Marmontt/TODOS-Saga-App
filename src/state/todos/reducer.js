import * as types from './types';

const initialState = {
    todos: []
}
const todosReducer = (state = initialState, action) => {
    switch (action.type) {
        default:
            return state;
    }
}

export default todosReducer;
