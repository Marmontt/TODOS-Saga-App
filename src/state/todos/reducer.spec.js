import reducer from '../todos/reducer'
import * as types from './types'

describe('todos reducer', () => {
    const initial_state = {
        chosenColor: '#eb716a',
        todos: []
    };

    it('should return initial state', () => {
        const state = reducer(undefined, {});
        expect(state).toMatchSnapshot();
    });
    it('should handle GET_TODOS_SUCCESS', () => {
        const todos = {color: '#fff', text: 'raq', fulfilled: false, index: 0};
        const getTodosSuccess = reducer({}, {type: types.GET_TODOS_SUCCESS, payload: todos});
        expect(getTodosSuccess).toMatchSnapshot();
    });
    it('should handle ADD_TODO_SUCCESS', () => {
        const todos = {text: 'raq', color: '#fff', id: 1, fulfilled: false};
        const addTodoSuccess = reducer(initial_state, {type: types.ADD_TODO_SUCCESS, payload: todos});
        expect(addTodoSuccess).toMatchSnapshot();
    });
    it('should handle SET_FULFILLED_SUCCESS', () => {
        const state = {
            chosenColor: '#eb716a',
            todos: [{text: 'raq', id: 0, color: '#fff', fulfilled: false}]
        };
        const fulfilled = true;
        const index = 0;
        const setFulfilledSuccess = reducer(state, {
            type: types.SET_FULFILLED_SUCCESS,
            payload: {index, fulfilled}
        });
        expect(setFulfilledSuccess).toMatchSnapshot();
    });
});
