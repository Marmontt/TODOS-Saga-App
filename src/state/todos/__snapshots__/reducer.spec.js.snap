// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todos reducer should handle ADD_TODO_SUCCESS 1`] = `
Object {
  "chosenColor": "#eb716a",
  "todos": Array [
    Object {
      "color": "#fff",
      "fulfilled": false,
      "id": 1,
      "text": "raq",
    },
  ],
}
`;

exports[`todos reducer should handle GET_TODOS_SUCCESS 1`] = `
Object {
  "todos": Object {
    "color": "#fff",
    "fulfilled": false,
    "index": 0,
    "text": "raq",
  },
}
`;

exports[`todos reducer should handle SET_FULFILLED_SUCCESS 1`] = `
Object {
  "chosenColor": "#eb716a",
  "todos": Array [
    Object {
      "color": "#fff",
      "fulfilled": true,
      "id": 0,
      "text": "raq",
    },
  ],
}
`;

exports[`todos reducer should return initial state 1`] = `
Object {
  "chosenColor": "#eb716a",
  "todos": Array [],
}
`;
